src/execution_plan/ops/op_update.c

```diff
@@ -20,7 +20,7 @@ OpBase* NewUpdateOp(GraphContext *gc, AST_Query *ast, ResultSet *result_set) {
     op_update->update_expressions_count = 0;
     op_update->entities_to_update_count = 0;
     op_update->entities_to_update_cap = 16; /* 16 seems reasonable number to start with. */
-    op_update->entities_to_update = malloc(sizeof(EntityUpdateEvalCtx) * op_update->entities_to_update_cap);
+    op_update->entities_to_update = malloc(sizeof(EntityUpdateCtx) * op_update->entities_to_update_cap);
```

Comment:
This allocation had formerly referred to the wrong struct type, which happened to be of the same size.